zingolib/src/testutils/chain_generics/fixtures.rs

```diff
@@ -540,3 +540,102 @@ where
         .await
     );
 }
+
+/// the simplest test that sends from a specific shielded pool to another specific pool. error variant.
+pub async fn shpool_to_pool_insufficient_error<CC>(
+    shpool: ShieldedProtocol,
+    pool: PoolType,
+    underflow_amount: u64,
+) where
+    CC: ConductChain,
+{
+    let mut environment = CC::setup().await;
+
+    let primary = environment.fund_client_orchard(1_000_000).await;
+    let secondary = environment.create_client().await;
+
+    let expected_fee = fee_tables::one_to_one(shpool, pool, true);
+    let secondary_fund = 100_000 + expected_fee - underflow_amount;
+    with_assertions::propose_send_bump_sync_all_recipients(
+        &mut environment,
+        &primary,
+        vec![(&secondary, Shielded(shpool), secondary_fund, None)],
+        false,
+    )
+    .await;
+
+    let tertiary = environment.create_client().await;
+
+    let ref_secondary: Arc<LightClient> = Arc::new(secondary);
+    let ref_tertiary: Arc<LightClient> = Arc::new(tertiary);
+
+    let tertiary_fund = 100_000;
+    assert_eq!(
+        from_inputs::propose(
+            &ref_secondary,
+            vec![(
+                ref_tertiary
+                    .wallet
+                    .get_first_address(pool)
+                    .unwrap()
+                    .as_str(),
+                tertiary_fund,
+                None,
+            )],
+        )
+        .await
+        .unwrap_err()
+        .to_string(),
+        format!(
+            "Insufficient balance (have {}, need {} including fee)",
+            secondary_fund,
+            tertiary_fund + expected_fee
+        )
+    );
+}
+
+/// the simplest test that sends from a specific shielded pool to another specific pool. also known as simpool.
+pub async fn to_pool_unfunded_error<CC>(pool: PoolType, try_amount: u64)
+where
+    CC: ConductChain,
+{
+    let mut environment = CC::setup().await;
+
+    let secondary = environment.create_client().await;
+    let tertiary = environment.create_client().await;
+
+    let ref_secondary: Arc<LightClient> = Arc::new(secondary);
+    let ref_tertiary: Arc<LightClient> = Arc::new(tertiary);
+
+    ref_secondary.do_sync(false).await.unwrap();
+
+    let mut expected_fee = fee_tables::one_to_one(Orchard, pool, true);
+
+    // why?
+    if pool == Shielded(Sapling) {
+        expected_fee -= 10_000
+    }
```

Comment:
we should not ignore these, lets fix fee tables asap but non-blocking I suppose

Reply:
it could be fee tables are correct but the zip317 iteraton returns the insufficient funds error before all input / outputs are taken into account

Reply:
can you add a TODO please

Reply:
i had assumed orchard in the fee tables. now the input can take None. (which is kinda silly, but it fits the math for now)